blocks/grid/grid.js

```diff
@@ -1,48 +1,40 @@
 import ComponentBase from '../../scripts/component-base.js';
-import { stringToJsVal } from '../../scripts/libs.js';
-import component from '../../scripts/init.js';
+import { flat, stringToJsVal } from '../../scripts/libs.js';
 
 export default class Grid extends ComponentBase {
-  static observedAttributes = [
-    'data-level',
-    'data-height',
-    'data-width',
-    'data-reverse',
-    'data-columns', // value can be any valid css value or a number which creates as many equal columns
-    'data-rows', // value can be any valid css value or a number which creates as many equal rows
-    'data-auto-columns',
-    'data-auto-rows',
-    'data-areas',
-    'data-justify-items',
-    'data-align-items',
-    'data-justify-content',
-    'data-align-content',
-  ];
-
-  nestedComponentsConfig = {};
+  // only one attribute is observed rest is set as css variables directly
+  static observedAttributes = ['data-reverse'];
 
   attributesValues = {
     all: {
-      data: {
-        level: 1,
+      grid: {
+        gap: '20px',
       },
     },
   };
 
-  setDefaults() {
-    super.setDefaults();
-  }
-
-  get gridItems() {
-    return [...this.children];
-  }
-
-  onAttributeHeightChanged({ oldValue, newValue }) {
-    this.setStyleProp('height', oldValue, newValue);
+  // use `grid` item as action from component base and apply those as css variables
+  // dinamic from {@link ../scripts/component-base.js:runConfigsByViewports}
+  // EG ${viewport}-grid-${attr}-"${value}"
+  applyGrid(grid) {
+    const f = flat(grid);
+    Object.keys(f).forEach((key) => {
+      this.style.setProperty(`--grid-${key}`, f[key]);
+    });
   }
 
-  onAttributeWidthChanged({ oldValue, newValue }) {
-    this.setStyleProp('width', oldValue, newValue);
+  // for backwards compatibility
```

Comment:
do we need to be compatible with anything? I would vouch for breaking change and fix content asap after merging if that's needed

Reply:
Just for current content. to work as needed.

Reply:
That way we maintain all properties set without grid prefixing grid, and avoiding having a callback for each attribute to change the variables